web/src/message_lists.ts

```diff
@@ -112,7 +104,7 @@ export function update_recipient_bar_background_color(): void {
 }
 
 export function save_pre_narrow_offset_for_reload(): void {
-    if (current === undefined) {
+    if (!current?.preserve_rendered_state) {
         return;
     }
```

Comment:
This check isn't obviously equivalent...

Reply:
Added this comment there:


```
    // Only save the pre_narrow_offset if the message list will be cached if user
    // switches to a different narrow, otherwise the pre_narrow_offset would just be when
    // user switches to a different narrow. In case of a reload, offset for the current
    // message is captured and restored by `reload` library.
```

Reply:
I think I'm still a bit confused. Reading c4bb181056ab0ac40c9340f460039bc4cda502f0 and 93ce1fa95ca3515c47db76c46b52137f58532769, I feel like either this function is badly named or it's not doing what I'd thinkg. Is `for_reload` about "for when we load this view again", not for reloading the browser?

Reply:
Opened #29871